common/scala/src/main/scala/whisk/core/database/ArtifactStore.scala

```diff
@@ -129,6 +129,11 @@ trait ArtifactStore[DocumentAbstraction] {
   protected[core] def readAttachment[T](doc: DocInfo, name: String, sink: Sink[ByteString, Future[T]])(
     implicit transid: TransactionId): Future[(ContentType, T)]
 
+  /**
+   * Deletes an attachments linked to given document
+   */
+  protected[core] def deleteAttachments[T](doc: DocInfo)(implicit transid: TransactionId): Future[Boolean]
```

Comment:
Can you clarify: Create one or all attachements linked?

Reply:
It deletes all linked attachments. See #3356 for more details on the API part

Reply:
>  +   * Deletes an attachments linked to given document 

-> Deletes all attachments linked to a given document.